val.py

```diff
@@ -310,6 +310,7 @@ def run(data,
     if not training:
         s = f"\n{len(list(save_dir.glob('labels/*.txt')))} labels saved to {save_dir / 'labels'}" if save_txt else ''
         LOGGER.info(f"Results saved to {colorstr('bold', save_dir)}{s}")
+        Path(LOGGER.handlers[0].baseFilename).rename(save_dir / 'log.txt')  # move log to save_dir
```

Comment:
Just ran into an issue because of this.
The loggers filehandler should be closed before moving that file.

```suggestion
        import logging
        LOGGER.handlers[0].close()
        Path(LOGGER.handlers[0].baseFilename).rename(save_dir / 'log.txt')  # move log to save_dir
        LOGGER.removeHandler(LOGGER.handlers[0])
        LOGGER.addHandler(logging.FileHandler('log.txt', 'w', delay=True))
```